### ISSUE TITLE
Handle Github API limits for projects with high request volume

### ISSUE THREAD

**Issue by pippinlee** [pippinlee/get-issues#20]
We should be aware of the problem of running `get-issues` for a public repo that doesn't require auth. Although it is likely our most common use case by a long shot Github's non-authed API has a rate limit that pretty easy to hit:
- if a public repo has 100s of open issues and issue comments this will likely hit the rate limit
- if a user is running `get-issues` on many projects they'll hit Github's rate limit

There are a few solutions to this:
- we could make `get-issues` always require auth, even for public repos. This is a sound choice, but means we add one extra barrier to people getting their issues. The majority of use will be through smallish public repos, so have `get-issues` be a one step command would be important, for our most common use case
- we could add logic that checks the api requests left, and if the rate limit will be hit before all issues have been downloaded locally, we can ask for auth. This would avoid us stopping the downloading of issues locally and asking for user auth which could get messy when keeping with state of issues downloaded. 

TLDR: This likely isn't a big issue because most use of `get-issues` is for <100 issue projects that are public, bit it's something we may hear about. Least work path: we just suggest people fake switch IP's to get around rate-limiting.
